### PR TITLE
CE-1658 Insights Popular Pages

### PR DESCRIPTION
**Pull request by kamilkoterba**
Added new Insights page using new refactored code
https://wikia-inc.atlassian.net/browse/CE-1658

Translations are already there.

@Wikia/community-engineering

### CODE REVIEW

File: extensions/wikia/InsightsV2/InsightsHelper.php [hunk 1]
```diff
@@ -90,6 +90,11 @@ public static function getInsightsPages() {
 			$dynamicInsights[InsightsFlagsModel::INSIGHT_TYPE] = 'InsightsFlagsModel';
 		}
 
+		/* Add PopularPages insight */
+		if ( !empty( $wgEnableInsightsPopularPages ) ) {
```
Comment (adamkarminski) [inline extensions/wikia/InsightsV2/InsightsHelper.php:94]: I'd move it above Flagged pages.

File: extensions/wikia/InsightsV2/specials/PopularPages.class.php [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	 * Returns an array with list of pages on wikia.
+	 *
+	 * @param bool $limit Only for consistency
+	 * @param bool $offset Only for consistency
+	 * @return array
+	 */
+	public function reallyDoQuery( $limit = false, $offset = false ) {
+		global $wgContentNamespaces;
+
+		$dbr = wfGetDB( DB_SLAVE, [ $this->getName(), __METHOD__, 'vslow' ] );
```
Comment (adamkarminski) [inline extensions/wikia/InsightsV2/specials/PopularPages.class.php:94]: It does not seem `vslow` :)

File: extensions/wikia/InsightsV2/specials/PopularPages.class.php [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+		$contentPages = ( new WikiaSQL() )
+			->SELECT( 'page_id', 'page_title', 'page_namespace' )
+			->FROM( 'page' )
+			->WHERE( 'page_namespace' )->IN( $wgContentNamespaces )
+			->runLoop( $dbr, function ( &$contentPages, $row ) {
```
Comment (adamkarminski) [inline extensions/wikia/InsightsV2/specials/PopularPages.class.php:100]: There are over 100k articles on some wikias. I think that a preliminary selection should happen at this stage. Especially given that you do not order the articles so may end up with first 1000 articles alphabetically sorted by PVs and not 1000 top articles.

**Comment by kamilkoterba** [comment]
moved here https://github.com/Wikia/app/pull/9417

**Pull request closed without merging**